ui/src/main/js/page/channel/azure/AzureBoardTable.js

```diff
@@ -57,6 +57,30 @@ const AzureBoardTale = ({ readonly, allowDelete }) => {
             sortOrder: data?.mutatorData?.direction
         }
     });
+    const [showModal, setShowModal] = useState(false);
+    const [statusMessage, setStatusMessage] = useState();
+    const [modalData, setModalData] = useState();
+
+    useEffect(() => {
+        // If a user authenticates via OAuth, OAuth will redirect us back to Alert with a url path similar to the following: 
+        // `alert/channels/azure_boards/edit/{id}`.  If an ID is present as well as the string 'edit', we can assert that 
+        // the modal should reopen with the data relevant to that ID.
+
+        // split the url to determine if edit is present
+        const parsedUrlArray = location.pathname.split('/');
+
+        if (parsedUrlArray.includes('edit')) {
+            // obtain the id of the azure board that OAuth just authenticated
+            const modalDataID = parsedUrlArray.slice(-1)[0];
+            // filter the table data and set the data for the modal to the one that matches the id from the line above
+            data.models.forEach(model => {
+                if (model.id === modalDataID) {
+                    setModalData(model);
+                    setShowModal(true);
+                }
+            })
```

Comment:
Does the "data" contain the page of azure boards configurations or ALL azure boards configurations?

Reply:
`data` here is all of the azure boards configuration that the store has.  What this is doing is it's taking the id out of the url (hacky method but it's our only option), then sifting through the data in the store to find a match on the id.  If it finds that match, it sets the data intended to be displayed in this modal (`modalData`).